dpdk/src/queue/rx.rs

```diff
@@ -50,25 +53,38 @@ pub struct RxQueueConfig {
     pub num_descriptors: u16,
     /// The socket preference for the rx queue.
     pub socket_preference: socket::Preference,
-    /// The low-level configuration of the rx queue.
-    pub config: (), // TODO
+    /// Hardware offloads to use
+    pub offloads: RxOffload,
     /// The memory pool to use for the rx queue.
     pub pool: mem::PoolHandle,
 }
 
 /// Error type for receive queue configuration failures.
-#[derive(Debug)]
+#[derive(Debug, thiserror::Error)]
 pub enum ConfigFailure {
-    /// The device has been removed.
-    DeviceRemoved(errno::Errno),
-    /// Invalid arguments were passed to the receive queue configuration.
-    InvalidArgument(errno::Errno),
-    /// Memory allocation failed.
-    NoMemory(errno::Errno),
-    /// An unexpected (i.e. undocumented) error occurred.
-    Unexpected(errno::Errno),
-    /// The socket preference setting did not resolve a known socket.
-    InvalidSocket(errno::Errno),
+    #[error("The device has been removed")]
+    DeviceRemoved(Errno),
+    #[error("Invalid arguments were passed to the receive queue configuration")]
+    InvalidArgument(Errno),
+    #[error("Memory allocation failed")]
+    NoMemory(Errno),
+    #[error("An unexpected (i.e. undocumented) error occurred")]
```

Comment:
Nit: or just “An undocumented error occurred”.

Also it would seem more natural to move that one to the end of the list.

Reply:
Fair enough

Reply:
Resolved I think

Reply:
You kept both “unexpected” and “undocumented”, which I think is a lot for an error message. But doesn't matter much, thanks for moving it down the list.

There's another occurrence of an enum where the `unexpected` error could be moved to the end (`enum ConfigFailure` in tx.rs). I leave it to you to see if you want to move it as well.

Reply:
will fix